### PR TITLE
QuarkusComponentTest: initial support for ConfigMapping

### LINKED ISSUE #36373: Support for non-mock `@ConfigMapping` in `@QuarkusComponentTest`
### Description

It would be nice if config mappings (`@ConfigMapping`-annotated interfaces) could be supported in `@QuarkusComponentTest`, either by default or as an opt-in feature.

Currently they seem to always be mocked, while I'd like them to be handled as they would in a normal application.

For example, if I have a component like this:

```
@ApplicationScoped 
public class Foo {

    @Inject
    Charlie charlie; 

    @Inject
    FooConfig config;

    @ConfigMapping(prefix = "foo")
    interface FooConfig {
        boolean bar();
    }

    public String ping() {
        return config.bar() ? charlie.ping() : "nok";
    }
}


```

And a test like this:

```
@QuarkusComponentTest 
@TestConfigProperty(key = "foo.bar", value = "true") 
public class FooTest {

    @Inject
    Foo foo; 

    @InjectMock
    Charlie charlieMock; 

    @Test
    public void testPing() {
        Mockito.when(charlieMock.ping()).thenReturn("OK"); 
        assertEquals("OK", foo.ping());
    }
}
```

... then the test won't pass because `FooConfig` is always injected as a mock, and as a result `config.bar()` returns `false` by default.

Using `new QuarkusComponentTestExtension(Foo.FooConfig.class)` doesn't seem to help.

### Implementation ideas

Quote from @mkouba (taken out of context, as all good quotes are):

> it should be doable ;-)

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #36373

### CODE REVIEW

File: test-framework/junit5-component/src/test/java/io/quarkus/test/component/config/ConfigMappingTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    @ConfigMapping(prefix = "foo")
+    interface FooConfig {
+
+        boolean bar();
+
+        @WithDefault("loom")
+        String baz();
+
+        String oof();
+    }
```
Comment (yrodiere) [inline test-framework/junit5-component/src/test/java/io/quarkus/test/component/config/ConfigMappingTest.java:55]: I suppose nested config mappings do not need extra care and are handled by Smallrye Config itself? I.e. if you had this:

```
    @ConfigMapping(prefix = "foo")
    interface FooConfig {

        boolean bar();

        @WithDefault("loom")
        String baz();

        String oof();

        Fii fii();

        interface Fii {
            String bar();
        }
    }
```

... then I guess `Fii` would not need to be handled explicitly by your code?

**Review by yrodiere — APPROVED** [review]
Wow that was fast, thanks!

This looks good to me. I didn't know we could set properties per test method, that should come in handy as well.

Just one comment, but it might be irrelevant.

File: test-framework/junit5-component/src/test/java/io/quarkus/test/component/config/ConfigMappingTest.java (hunk 1 above)
Comment (mkouba) [reply test-framework/junit5-component/src/test/java/io/quarkus/test/component/config/ConfigMappingTest.java:55]: That's my understanding too. But I can add a test to make sure it really works ;-).

**Comment by mkouba** [comment]
>  I didn't know we could set properties per test method, that should come in handy as well.

:+1: yep, that was a feature request from one of our users.

Keep in mind, that it only works if the test instance lifecycle is `Lifecycle#PER_METHOD` (default).

File: test-framework/junit5-component/src/test/java/io/quarkus/test/component/config/ConfigMappingTest.java (hunk 1 above)
Comment (mkouba) [reply test-framework/junit5-component/src/test/java/io/quarkus/test/component/config/ConfigMappingTest.java]: Test added.

**Review by yrodiere — APPROVED** [review]

**Pull request merged**